Patch-Source: https://github.com/bugaevc/wl-clipboard/commit/abfd513ecceef5f33632e0f63daea92f9bbe08e2
--
From abfd513ecceef5f33632e0f63daea92f9bbe08e2 Mon Sep 17 00:00:00 2001
From: Sergey Bugaev <bugaevc@gmail.com>
Date: Fri, 29 Sep 2023 14:48:22 +0300
Subject: [PATCH] Fix assertion failure

If we try to paste when there is an offer that has no MIME types offered,
we'd fail when trying to report it to the user. Make sure to handle this
case explicitly, and report it the same way as there no being any offer
at all.
---
 src/wl-paste.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/wl-paste.c b/src/wl-paste.c
index d9a405a..dee2fad 100644
--- a/src/wl-paste.c
+++ b/src/wl-paste.c
@@ -210,7 +210,13 @@ static int run_paste_command(int stdin_fd, const char *clipboard_state) {
     return 1;
 }
 
-static void complain_no_suitable_type() {
+static void complain_no_suitable_type(const struct types *types) {
+    if (types->any == NULL) {
+        /* Report this the same way as
+         * there being no offer at all.
+         */
+        bail("Nothing is copied");
+    }
     fprintf(stderr, "Clipboard content is not available as ");
     if (options.explicit_type != NULL) {
         fprintf(stderr, "requested type \"%s\"\n", options.explicit_type);
@@ -277,7 +283,7 @@ static void selection_callback(struct offer *offer, int primary) {
             offer_destroy(offer);
             return;
         }
-        complain_no_suitable_type();
+        complain_no_suitable_type(&types);
     }
 
     /* Never append a newline character to binary content */
